client/signup/main.jsx

```diff
@@ -525,6 +525,7 @@ class Signup extends React.Component {
 		const isDomainsForSiteEmpty =
 			this.props.isLoggedIn &&
 			this.props.signupDependencies.siteSlug &&
+			this.props.signupDependencies.siteSlug !== 'no-site' &&
```

Comment:
👍

Reply:
❤️ Thanks @kwight! This solves the JS error I was seeing. The checkout page is still not loaded, so I guess the signup code is still waiting for a new site to be created before doing the redirect. Will keep investigating to see if I can identify where.